### PR TITLE
Update libgit2 natives to 1.0.160

### PR DESCRIPTION
**Pull request by ethomson**
Update libgit2 to native version 1.0.160 (https://github.com/libgit2/libgit2/commit/baa87dfc5033364ab5ac71099725f8f6c3e6361c).

### CODE REVIEW

**Comment by txdv** [comment]
How did I miss this...

Looks, good, are you planning on exposing the new APIs in a different commit?

File: LibGit2Sharp/Core/NativeMethods.cs [hunk 1]
```diff
@@ -989,7 +976,7 @@ static NativeMethods()
 
         /* Push network progress notification function */
         internal delegate int git_push_transfer_progress(uint current, uint total, UIntPtr bytes, IntPtr payload);
-        internal delegate int git_packbuilder_progress(int stage, uint current, uint total, IntPtr payload);
+        internal delegate int git_packbuilder_progress(int stage, UInt32 current, UInt32 total, IntPtr payload);
```
Comment (txdv) [inline LibGit2Sharp/Core/NativeMethods.cs]: Is this really needed? uint is always a synonym for System.UInt32

Comment (ethomson) [inline LibGit2Sharp/Core/NativeMethods.cs]: Right, probably not.  Thanks!

**Comment by ethomson** [comment]
> Looks, good, are you planning on exposing the new APIs in a different commit?

I don't have any plans to.  Is there something in particular that you're interested in?  Is that something that you'd want to tackle?

**Comment by txdv** [comment]
Yeah, I would like to tackle it(exposnig all the new APIs) beginning with (Set|Get)UserAgent . Actually, I already started doing that, but I ran against https://github.com/libgit2/libgit2sharp/pull/1339 and kind of lost track...

**Comment by carlosmn** [comment]
Unless we've deprecated something, in which case we should immediately provide the new way of doing it, I'd rather not conflate the binaries update with exposing new functionality.

File: LibGit2Sharp/packages.config [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="LibGit2Sharp.NativeBinaries" version="1.0.157" targetFramework="net4" allowedVersions="[1.0.157]" />
+  <package id="LibGit2Sharp.NativeBinaries" version="1.0.160" targetFramework="net40" />
```
Comment (carlosmn) [inline LibGit2Sharp/packages.config]: I think this should still have a `allowedVersions` to stop it from trying to "update" to an incompatible version of the binaries.

Comment (ethomson) [inline LibGit2Sharp/packages.config]: Fixed.

File: LibGit2Sharp/Core/NativeMethods.cs (hunk 1 above)
Comment (ethomson) [inline LibGit2Sharp/Core/NativeMethods.cs]: Fixed.

**Pull request merged**